extensions/blocks/dialogue/components/time-stamp-control.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Dropdown,
+	Button,
+	__experimentalNumberControl as NumberControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+function validateValue( val, max ) {
+	return Math.max( 0, Math.min( val, max ) );
+}
+
+const timeStampMap = [ 'hour', 'min', 'sec' ];
+
+/**
+ * Helper function to parse and stringify the time stamp,
+ * in the HH:MM:SS format.
+ * `HH` is optional, meaning when it's empty,
+ * the string won't contain it, returning MM:SS.
+ *
+ * @param {object} typeValue - { type: value } pair value of the time stamp.
+ * @param {Array} smh - current [ second, menute, hour ] array values.
+ * @returns {string} HH:MM:SS format.
+ */
+function setTimeStampValue( typeValue, smh ) {
```

Comment:
Do you think this could be extracted into an utility, with some tests along with it?

Reply:
yes, in fact, we think it could be implemented straight in core. Clearly in a follow-up PR.
p1607641100481700?thread_ts=1607640662.481300&cid=C45SNKV4Z-slack-